vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -144,6 +149,11 @@ func (u *updater) RunOnce() {
 			if !evictionLimiter.CanEvict(pod) {
 				continue
 			}
+			err := u.evictionRateLimiter.Wait(ctx)
```

Comment:
`Wait` always asks for one token, so it doesn't make much sense to set the burst rate to anything else than 1. If I'm correct then we should remove the `eviction-rate-limit-burst` parameter.

Reply:
Without knowing the autoscaler well I'm a little bit worried if `Wait` will block here for long time (e.g. minutes). The state of the cluster might change significantly during that time.

Reply:
An idea for solving the Wait()/blocking problem:
1. Let's collect all pods to be evicted in a list
2. Randomize the list (to give all pods a fair chance to be evicted in a RunOnce run)
3. Go through the list and evict a pod if `u.evictionRateLimiter.Allow()` returns true.
4. If `u.evictionRateLimiter.Allow()` returns false then return from the RunOnce function. Any remaining pods to be evicted will be retried in the next RunOnce run

Reply:
Thanks for you input 🙂 

I'm not really sure why the Wait is an issue here :
- A context with timeout it passed to the Wait function so the Wait will be cancelled if the delay is larger than the tick interval.
- As far as I understand it. The state of the cluster is updated by the recommender, and the admission controller applies it. The updater is only responsible for eviction here, and when the pod restarts, the last value of the VPA are applied by the admission controller.

In your proposed solution, we would hit the rate limiter quickly and exit. So we would "waste" n seconds of the ticker loop doing nothing. Running on fresher data is an excellent idea, but the only option I see would be to refresh the pod list between every Wait, which doesn't seem ideal to me.

Reply:
The default tick for the updater is 1 minute, so as @guillaumebreton writes, the Wait will be canceled after that time at most (unless someone uses a custom tick). So we can basically look at info that is stale up to 1 minute, which is also true if updater has a lot of work to do. I'm inclined to agree this solution is acceptable. 

As to randomizing the list of pods to evict - the updater has a priority mechanism to select which pods to evict first based on multiple factors (including how far away they are from their recommended resources). Shuffling the list wouldn't play well with that.